app/hotels/src/singleHotel/roomList/RoomRow.js

```diff
@@ -52,22 +51,26 @@ export class RoomRow extends React.Component<Props> {
   };
 
   getIdAndMaxPersons = () => {
-    const id = idx(this.props, _ => _.availableRoom.id);
-    const maxPersons = idx(this.props, _ => _.availableRoom.room.maxPersons);
+    const id = this.props?.availableRoom?.id;
+    const maxPersons = this.props?.availableRoom?.room?.maxPersons;
     return { id, maxPersons };
   };
 
   openGallery = () => {
     const { availableRoom } = this.props;
-    const photosArray = idx(availableRoom, _ => _.room.roomPhotos) || [];
+    const photosArray = availableRoom?.room?.roomPhotos ?? [];
 
-    const photos = photosArray.map(photo => ({
-      key: idx(photo, _ => _.id) || '',
-      highRes: idx(photo, _ => _.highResUrl) || '',
-      lowRes: idx(photo, _ => _.lowResUrl) || '',
-    }));
+    const photos = photosArray.map(photo => {
```

Comment:
```suggestion
    const photos = photosArray.map(photo => {
      const key =  photo?.id ?? '';
      const highRes = photo?.highResUrl ?? '';
      const lowRes = photo?.lowResUrl ?? highRes; // Fallback to highResUrl since stay22 do not provide lowResUrl
      
      return {
        key,
        highRes,
        lowRes,
      };
    });
```

Reply:
Sorry, that does not seem to replace the lines below... But I think it's better that all the logic is in the `return` statement or before the `return` statement, and not mixed like it is now. Thoughts?

Reply:
makes sense